app/gui/src/controller/searcher.rs

```diff
@@ -725,24 +537,24 @@ impl Searcher {
     /// This function should be called each time user modifies Searcher input in view. It may result
     /// in a new action list (the appropriate notification will be emitted).
     #[profile(Debug)]
-    pub fn set_input(&self, new_input: String) -> FallibleResult {
-        debug!("Manually setting input to {new_input}.");
-        let parsed_input = ParsedInput::new(new_input, self.ide.parser())?;
-        let old_expr = self.data.borrow().input.expression.repr();
-        let new_expr = parsed_input.expression.repr();
-
-        self.data.borrow_mut().input = parsed_input;
-        self.invalidate_fragments_added_by_picking();
-        let expression_changed = old_expr != new_expr;
-        if expression_changed {
+    pub fn set_input(&self, new_input: String, cursor_position: Byte) -> FallibleResult {
+        debug!("Manually setting input to {new_input} with cursor position {cursor_position}");
+        let parsed_input = input::Input::parse(self.ide.parser(), new_input, cursor_position);
+        let new_context = parsed_input.context().map(|ctx| ctx.into_ast().repr());
+        let old_input = mem::replace(&mut self.data.borrow_mut().input, parsed_input);
+        let old_context = old_input.context().map(|ctx| ctx.into_ast().repr());
+
+        self.invalidate_picked_suggestions();
+        let context_changed = old_context != new_context;
```

Comment:
I'm not sure if we're doing the correct thing here. Previous code compared expressions, now we compare access chains, and they usually `None`. So we are reloading the list far less frequently now.

Reply:
This is correct. Before we were filtering the components by expected argument type (and it depended on the left side of the expression), but now the filtering depends only on accessor chain (i.e. the context).